src/cloudai/configurator/base_agent.py

```
@@ -111,3 +112,38 @@ def update_policy(self, _feedback: Dict[str, Any]) -> None:
+            observation, reward, done, *_ = self.env.step(action)
+            self.update_policy(
+                {
+                    "trial_index": step,
+                    "value": reward,
+                    "observation": observation,
+                    "prev_observation": prev_observation,
+                    "action": action,
+                    "done": done,
+                }
+            )
+            logging.info(f"Step {step}: Observation: {[round(obs, 4) for obs in observation]}, Reward: {reward:.4f}")
```

Ruff (0.15.15)

[warning] Logging statement uses f-string

(G004)